src/wechaty_puppet_service/puppet.py

```diff
@@ -898,7 +898,12 @@ def _init_puppet(self):
             log.debug('get puppet ip address : <%s>', data)
             self.options.end_point = '{ip}:{port}'.format(**data)
 
-        if ping(self.options.end_point) is False:
+        if ':' not in self.options.end_point:
```

Comment:
`:` is not required for the valid endpoint which can be the service name or url. Eg: `wechaty-service` in docker network or `http://api.wechaty-service.com` .

Reply:
i got it

Reply:
we can make a rule: 

WECHATY_PUPPET_SERVICE_ENDPOINT must be written as {Host/Domain Name}:{port}

if use default port, we must write it. Eg: for 'http', we use '80', for 'https', we use '443'

Reply:
we must follow the identical rules when we write the `WECHATY_PUPPET_SERVICE_ENDPOINT`, otherwise there will be many cases that we can't deal with.

Reply:
I believe there must be some bugs in ping code. but port is not required in `ping` function. Can you reproduce your problem? And can you fix it after add port support ?

Reply:
the following code is good for me:
```python
from ping3 import ping
ping("www.baidu.com")
ping("45.113.192.102") # the ip of www.baidu.com
```

how do you think about it ?

Reply:
yeah, as you said, port is not required in `ping` function. 

Here the bug is: `if ping(self.options.end_point) is False:`

1.My Problem:
When I write my `WECHATY_PUPPET_SERVICE_ENDPOINT` like `127.0.0.1:8888`, the program raise an Exception: 
`wechaty_puppet.exceptions.WechatyPuppetConfigurationError: WechatyPuppetConfigurationError("can't not ping endpoint: [ip:port]", None, None)```

Reply:
my `WECHATY_PUPPET_SERVICE_ENDPOINT` can be written as `www.baidu.com:9999`

if I wang to ping host, i can write as `ping(self.options.end_point.split(':')[0])`, 

how do you think about it ?

Reply:
Yes, and what's more, after checking the details of `ping` fucntion and protocol, it can only check the valid status of server ip and can not check the accessibility of server port. 

So, I think there are more improvments in this situation:

Reply:
Yes, you mean that it needs `telnet`.

socket is a good way, now i'm looking for the methods about socket

Reply:
Great, looking forward for your latest commit. You can encapsulate it into `utils.py` module.

Reply:
There are some developers waiting for this pr, please complete the development and testing of this feature as soon as possible. Many thanks for your sharing and understanding.

Reply:
ok, i will make it soon

Reply:
I did a simple local test, no problem. please review my code.